### ISSUE TITLE
"config=gdb" build uses incorrect indices for 'Field2' in 'extract_slice_yz' (and others).

### ISSUE THREAD

**Issue by ekapadi** [SmileiPIC/Smilei#215]
The following crashes any namelist on a debug build using external fields:
```
void Field3D::extract_slice_yz( unsigned int ix, Field2D *slice )
 {
-    DEBUGEXEC( if( dims_[1]!=slice->dims_[1] ) ERROR( name << " : " <<  dims_[1] << " and " << slice->dims_[1] ) );
-    DEBUGEXEC( if( dims_[2]!=slice->dims_[2] ) ERROR( name << " : " <<  dims_[2] << " and " << slice->dims_[2] ) );
+    DEBUGEXEC( if( dims_[1]!=slice->dims_[0] ) ERROR( name << " : " <<  dims_[1] << " and " << slice->dims_[0] ) );
+    DEBUGEXEC( if( dims_[2]!=slice->dims_[1] ) ERROR( name << " : " <<  dims_[2] << " and " << slice->dims_[1] ) );
```
Pull request available...
(Please contact me directly. -- Thanks! )

**Comment by ekapadi** [comment]
This issue should definitely be marked as a [BUG].  I can't see any way to do that from here.

**Comment by beck-llr** [comment]
Hi
thanks for the reports and  for being willing to provide corrections. This is really appreciated !

Isn'it possible for you to use the pull request feature available here on git-hub ?

**Comment by jderouillat** [comment]
Thank you for your attention, the debug mode is not integrated in our CI tests, and apparently nobody didn't need to use it with external fields in 3D for now. The bug fix has been pushed here.

**Comment by ekapadi** [comment]
Hi:

Here's my situation:

   - I've forked the Smilei repo onto my local server.
   - I then created a branch to add my bug fix.

I read through all of the github documentation on pull requests.  It still
really isn't clear what repo I'm supposed to push my pull request to.  My
'git remote' (for these branches) reads: "origin
https://github.com/SmileiPIC/Smilei.git" (both for fetch / push), but
obviously I shouldn't have any authorization yet to push to that remote.

Further (and also obviously), I hit the "pull request" button, and there's
no branch showing up that I can indicated for github to take the pull
request from.

At my employer, we use a shared-repo model, and I do have authorization to
push (, I have a repo there <ekapadi: my github username>/<my employer's
name>).

It appears to me that I need some additional authorizations from your team
before I'll be able to contribute.  (Note that under my github username
'ekapadi', I don't have any remaining repos to spare for this purpose, so
hopefully that's not what you're expecting to enable this.)

Yours,

Kort

**Comment by mccoys** [comment]
On GitHub, the process is to "Fork" the repo. You can submit a pull request from that fork.

I suggest you fork from github on your own account. Then, you can make your changes in that fork. But if you do not want to copy all your changes, you can use your existing local copy and simply change the URL to the new fork:
```
git remote set-url origin https://github.com/user/FORK.git
```

After that, you should be able to do a pull request

**Comment by jderouillat** [comment]
Fixed from 254b5e53e7d8e496c8f1bb84744dfececbd53acd
